### PR TITLE
site: Added versioned access logging document

### PR DESCRIPTION
**Pull request by tsaarni**
Added information about 

* customizing text based access logs
* supported access log formatter extensions (there is only one so far)

This documentation is related to #3694 and #3576

Signed-off-by: Tero Saarni <tero.saarni@est.tech>

### CODE REVIEW

**Comment by tsaarni** [comment]
xref #3829

File: site/content/docs/main/config/access-logging.md
Comment (skriss) [inline site/content/docs/main/config/access-logging.md:7]:
```suggestion
By default, HTTP and HTTPS access logs are written to `/dev/stdout` by the Envoy containers and look like the following:
```

File: site/content/docs/main/config/access-logging.md
Comment (skriss) [inline site/content/docs/main/config/access-logging.md:39]: I'd add a note here that the format string needs to end in a `\n`

File: site/data/docs/main-toc.yml
Comment (skriss) [inline site/data/docs/main-toc.yml:43]: I'd move this up to just above "Annotations Reference" to leave the two reference pages at the bottom of this TOC section. Separately (not needed for this PR) we might want to move them into their own "Reference" section.

**Review by skriss — CHANGES_REQUESTED** [review]
Looks pretty good to me @tsaarni, just a couple small comments. It looks like this effectively replaces the existing guide, so should we remove it?

File: site/content/docs/main/config/access-logging.md
Comment (youngnick) [inline site/content/docs/main/config/access-logging.md:109]: As with my comment on #3694, I don't think that we should include this until it's included in the Envoy version that Contour supports.

**Review by youngnick — CHANGES_REQUESTED** [review]
Similarly to #3694, I think that we should either remove references to REQ_WITHOUT_QUERY until Envoy supports it, or hold this PR until then.

**Comment by tsaarni** [comment]
> It looks like this effectively replaces the existing guide, so should we remove it?

How should we cover the old versions with regards #3829 while moving information from guides?

Before removing the original document from site level guides, should the information be duplicated to old versions as well for clarity? And in that case, I guess we should also check what information applies to which version, e.g. JSON logs beginning from Contour 1.10.0.

**Comment by tsaarni** [comment]
> Similarly to #3694, I think that we should either remove references to REQ_WITHOUT_QUERY until Envoy supports it, or hold this PR until then.

Lets hold until Envoy 1.19.0 is released!

File: site/content/docs/main/config/access-logging.md
Comment (tsaarni) [reply site/content/docs/main/config/access-logging.md:38]: I've added note about `\n`.  I also linked the page about Envoy command operator explanations here in this chapter.

File: site/content/docs/main/config/access-logging.md
Comment (tsaarni) [reply site/content/docs/main/config/access-logging.md]: Fixed!

File: site/data/docs/main-toc.yml
Comment (tsaarni) [reply site/data/docs/main-toc.yml]: Fixed!

**Comment by tsaarni** [comment]
I noticed that in the JSON chapter, the links to Contour code (for the list of command operators in `envoySimpleOperators` and `envoyComplexOperators`) are bit out-of-sync.  If we wan't to keep linking to the code, maybe we should fix the git hash in links to a specific version or just link to the file, without deep linking to line numbers?

Also, I guess user would be interested in reading the Envoy docs what the command operators do. We could have a separate chapter that links to Envoy doc as explanation on what the operators do and then say that from available command operators, Contour supports X,Y and Z.  

The list of command operators is a bit moving target and even now Envoy has some command operators introduced in latest released Envoy that are not known by the current parser code. Maybe these should be added?

**Comment by skriss** [comment]
> I noticed that in the JSON chapter, the links to Contour code (for the list of command operators in `envoySimpleOperators` and `envoyComplexOperators`) are bit out-of-sync. If we wan't to keep linking to the code, maybe we should fix the git hash in links to a specific version or just link to the file, without deep linking to line numbers?

Yeah, I think either of these could work; I'd probably lean towards just linking to the file and describing which part to look at.

> Also, I guess user would be interested in reading the Envoy docs what the command operators do. We could have a separate chapter that links to Envoy doc as explanation on what the operators do and then say that from available command operators, Contour supports X,Y and Z.

Yeah, I think this is a good idea.

> The list of command operators is a bit moving target and even now Envoy has some command operators introduced in latest released Envoy that are not known by the current parser code. Maybe these should be added?

:+1: also probably a good idea, can file a separate issue to add the missing one.

**Comment by skriss** [comment]
> > It looks like this effectively replaces the existing guide, so should we remove it?
> 
> How should we cover the old versions with regards #3829 while moving information from guides?
> 
> Before removing the original document from site level guides, should the information be duplicated to old versions as well for clarity? And in that case, I guess we should also check what information applies to which version, e.g. JSON logs beginning from Contour 1.10.0.

These are all good points. I'm fine with adding this new doc for now without removing the guide, and then we can resolve these questions as part of #3829.

**Review by youngnick — APPROVED** [review]
Since the support for REQ_WITHOUT_QUERY is now in, I'm good with this going in.

**Comment by skriss** [comment]
@tsaarni I'll leave this open for now in case you can address the recent comments, but they can also be addressed in a follow-up if needed, we'll definitely get this in for the upcoming release.

**Review by sunjayBhatia — COMMENTED** [review]
content LGTM, pending the content that may be added based on other reviews

**Comment by skriss** [comment]
I filed #3920 for follow-ups, merging this as-is. Thanks @tsaarni!

**Review by skriss — APPROVED** [review]

**Pull request merged**
